### PR TITLE
Add test with zero in off-diagonal entry

### PR DESCRIPTION
**Pull request by blegat**
This test would have caught the OSQP issue fixed by https://github.com/oxfordcontrol/OSQP.jl/pull/90

- [x] Add comments
- [x] Add tests

### CODE REVIEW

File: src/Test/UnitTests/objectives.jl [hunk 1]
```diff
@@ -314,6 +314,50 @@ function solve_qp_edge_cases(model::MOI.ModelLike, config::TestConfig)
...
+            0.0,  # constant
+        ),
+    )
+    test_model_solution(
+        model,
+        config;
+        objective_value = 5.0,
+        variable_primal = [(x[1], 1.0), (x[2], 2.0)],
+    )
+end
```
Comment (odow) [inline src/Test/UnitTests/objectives.jl:359]: Needs
```
unittests["solve_qp_zero_offdiag"] = solve_qp_zero_offdiag
```

This could go in `solve_qp_edge_cases` so it doesn't break the tests of every solver who doesn't support QP? Or add a check for `supports`?

Comment (blegat) [reply src/Test/UnitTests/objectives.jl:359]: I tried adding it to `edge_case` but the bug in OSQP only occurs at `copy_to`, not if you do `MOI.set` so it didn't break OSQP.
